instance/import.go

```diff
@@ -0,0 +1,187 @@
+package instance
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"strconv"
+
+	errs "github.com/ONSdigital/dp-dataset-api/apierrors"
+	"github.com/ONSdigital/dp-dataset-api/models"
+	"github.com/ONSdigital/go-ns/audit"
+	"github.com/ONSdigital/go-ns/common"
+	"github.com/ONSdigital/go-ns/log"
+	"github.com/ONSdigital/go-ns/request"
+	"github.com/gorilla/mux"
+	"github.com/pkg/errors"
+)
+
+// UpdateObservations increments the count of inserted_observations against
+// an instance
+func (s *Store) UpdateObservations(w http.ResponseWriter, r *http.Request) {
+	ctx := r.Context()
+	vars := mux.Vars(r)
+	instanceID := vars["instance_id"]
+	insert := vars["inserted_observations"]
+	auditParams := common.Params{"instance_id": instanceID, "inserted_observations": insert}
+	logData := audit.ToLogData(auditParams)
+
+	if err := func() error {
+		observations, err := strconv.ParseInt(insert, 10, 64)
+		if err != nil {
+			log.ErrorCtx(ctx, errors.WithMessage(err, "update imported observations: failed to parse inserted_observations string to int"), logData)
+			return errs.ErrInsertedObservationsInvalidSyntax
+		}
+
+		if err = s.UpdateObservationInserted(instanceID, observations); err != nil {
+			log.ErrorCtx(ctx, errors.WithMessage(err, "update imported observations: store.UpdateObservationInserted returned an error"), logData)
+			return err
+		}
+
+		return nil
+	}(); err != nil {
+		if auditErr := s.Auditor.Record(ctx, UpdateInsertedObservationsAction, audit.Unsuccessful, auditParams); auditErr != nil {
+			err = auditErr
+		}
+		handleInstanceErr(ctx, err, w, logData)
+		return
+	}
+
+	s.Auditor.Record(ctx, UpdateInsertedObservationsAction, audit.Successful, auditParams)
+
+	log.InfoCtx(ctx, "update imported observations: request successful", logData)
+}
+
+// UpdateImportTask updates any task in the request body against an instance
+func (s *Store) UpdateImportTask(w http.ResponseWriter, r *http.Request) {
+
+	defer request.DrainBody(r)
+
+	ctx := r.Context()
+	vars := mux.Vars(r)
+	instanceID := vars["instance_id"]
+	auditParams := common.Params{"instance_id": instanceID}
+	logData := audit.ToLogData(auditParams)
+	defer r.Body.Close()
+
+	updateErr := func() *taskError {
+		tasks, err := unmarshalImportTasks(r.Body)
+		if err != nil {
+			log.ErrorCtx(ctx, errors.WithMessage(err, "failed to unmarshal request body to UpdateImportTasks model"), logData)
+			return &taskError{err, http.StatusBadRequest}
+		}
+
+		validationErrs := make([]error, 0)
+		var hasImportTasks bool
+
+		if tasks.ImportObservations != nil {
+			hasImportTasks = true
+			if tasks.ImportObservations.State != "" {
+				if tasks.ImportObservations.State != models.CompletedState {
+					validationErrs = append(validationErrs, fmt.Errorf("bad request - invalid task state value for import observations: %v", tasks.ImportObservations.State))
```

Comment:
would it be useful for errors like this explicitly say _why_ it's invalid? e.g. what values would have been acceptable

also seems like we're falling back to using concatenation/sprintf rather than structured log data?